dvc/remote/ssh/connection.py

```diff
@@ -311,3 +319,32 @@ def md5(self, path):
     def cp(self, src, dest):
         self.makedirs(posixpath.dirname(dest))
         self.execute("cp {} {}".format(src, dest))
+
+    @contextmanager
+    def open_max_sftp_channels(self):
+        try:
+            channels = []
+
+            with self.mute_paramiko():
+                while True:
+                    try:
+                        channels.append(self._ssh.open_sftp())
+                    except paramiko.ssh_exception.ChannelException:
```

Comment:
are we exhausting all SSH connections on the server this way? or just a number of channels per connections/

my questions is - how this greedy aggressive strategy affects other potential SSH clients on the server?

Reply:
good question, @shcheklein!

> how this greedy aggressive strategy affects other potential SSH clients on the server?

Running `htop` on the server while running `dvc status -r -ssh` showed no significant resource consumption:

Before:
![2019-06-18-15:02:36](https://user-images.githubusercontent.com/7363250/59715750-27e92680-91da-11e9-851b-f7c27d6a1c4c.png)


During `dvc status`:
![2019-06-18-15:03:09](https://user-images.githubusercontent.com/7363250/59715769-2fa8cb00-91da-11e9-8736-354276091c98.png)


And I could still open more SSH and SFTP connections without any error.

So, based on this observations, I'd say that it is not that aggressive to become a potential problem. Although, it raises the question of why paramiko is limiting us to create more than 10 jobs :thinking: